### PR TITLE
Add OAuth2 (non-metaverse) Logic to Interface & Domain

### PR DESCRIPTION
**Pull request by kasenvr**

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR566/Vircadia-Alpha-PR566-e412d9b_a5d0c80c.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR566/Vircadia-Alpha-PR566-e412d9b.tar_a5d0c80c.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR566/Vircadia-Alpha-PR566-e412d9b.tar_a5d0c80c.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR566/Vircadia-Alpha-PR566-e412d9b_a5d0c80c.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR566/Vircadia-Alpha-PR566-e412d9b_a5d0c80c.exe

File: libraries/networking/src/DomainHandler.h [hunk 1]
```diff
@@ -199,7 +199,8 @@ class DomainHandler : public QObject {
         Unknown,
         ProtocolMismatch,
         LoginError,
-        NotAuthorized,
+        NotAuthorizedMetaverse,
+        NotAuthorizedDomain,
```
Comment (ctrlaltdavid) [inline libraries/networking/src/DomainHandler.h:203]: The new reason should be appended to the end of the list in order to maintain compatibility with previous releases on the same prototol and any scripts using the value. I'll move its location in another PR.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR566/Vircadia-Alpha-PR566-f2cdf2e_bcb469a2.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR566/Vircadia-Alpha-PR566-f2cdf2e.tar_bcb469a2.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR566/Vircadia-Alpha-PR566-f2cdf2e.tar_bcb469a2.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR566/Vircadia-Alpha-PR566-f2cdf2e_bcb469a2.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR566/Vircadia-Alpha-PR566-f2cdf2e_bcb469a2.exe

**Comment by ctrlaltdavid** [comment]
These changes on the `oauth2-wordpress` feature branch and shouldn't be merged into master yet.

**Pull request closed without merging**